tests/test_trimmer.py

```diff
@@ -16,24 +18,49 @@ def setUp(self):
     def tearDown(self):
         pass
 
+    def make_wcs(self, instcat_file, det_name):
+        obs_md = imsim.OpsimMetaDict(instcat_file)
+        boresight = galsim.CelestialCoord(ra=obs_md['rightascension'] * galsim.degrees,
+                                          dec=obs_md['declination'] * galsim.degrees)
+        rotTelPos = obs_md['rottelpos'] * galsim.degrees
+        obstime = astropy.time.Time(obs_md['mjd'], format='mjd', scale='tai')
+        band = obs_md['band']
+
+        builder = imsim.BatoidWCSBuilder()
+        return builder.makeWCS(boresight, rotTelPos, obstime, det_name, band)
+
     def test_InstCatTrimmer(self):
         """Unit test for InstCatTrimmer class."""
-        instcat = os.path.join(os.environ['IMSIM_DIR'], 'tests',
-                               'tiny_instcat.txt')
-        sensor = 'R:2,2 S:1,1'
+        instcat_file = 'tiny_instcat.txt'
+        sensor = 'R22_S11'
+        wcs = self.make_wcs(instcat_file, sensor)
+
+        # Note: some objects in instcat are up to ~600 pixels off the image.
+        # So need to use a largish edge_pix to keep them all.
+        instcat = imsim.InstCatalog(instcat_file, wcs, edge_pix=1000, skip_invalid=False)
+        self.assertEqual(instcat.nobjects, 24)
 
-        # Check the application of minsource.
-        objs = desc.imsim.InstCatTrimmer(instcat, [sensor], minsource=10)
-        self.assertEqual(len(objs[sensor]), 24)
+        # With the default edge_pix=100, only 17 make the cut.
+        instcat = imsim.InstCatalog(instcat_file, wcs, skip_invalid=False)
+        self.assertEqual(instcat.nobjects, 17)
 
-        objs = desc.imsim.InstCatTrimmer(instcat, [sensor], minsource=12)
-        self.assertEqual(len(objs[sensor]), 0)
+        # Check the application of min_source.
+        instcat = imsim.InstCatalog(instcat_file, wcs, edge_pix=1000, min_source=10,
+                                    skip_invalid=False)
+        self.assertEqual(instcat.nobjects, 24)
+
+        instcat = imsim.InstCatalog(instcat_file, wcs, edge_pix=1000, min_source=12,
+                                    skip_invalid=False)
+        self.assertEqual(instcat.nobjects, 0)
 
         # Check various values of chunk_size.
-        for chunk_size in (5, 10, 100):
-            objs = desc.imsim.InstCatTrimmer(instcat, [sensor], minsource=None,
-                                             chunk_size=chunk_size)
-            self.assertEqual(len(objs[sensor]), 24)
+        # XXX: We don't use chunking anymore.  If we revisit this and put it back in,
+        #      then we can re-enable this test.
```

Comment:
Right.  I meant if we decided that the new code should be chunking like you did in the trimmer, then we might want a version of this test, modified to test the new code. I guess I'm also kind of prodding that we recall why we did chunking in the old case and think about whether it would make sense to add something similar now.

Reply:
IIRC, the original reason was to avoid using up a lot of memory when reading in the instance catalogs (which covered the full focal plane) since most of the entries would be omitted once the objects that landed on the CCD were identified.

Reply:
OK, I've removed this.  In the new code, we already have the WCS as we're reading in the file, so there's no overhead of keeping any objects in memory that won't end up being kept.